pkg/prom/instance/host_filter.go

```diff
@@ -58,13 +63,41 @@ func NewHostFilter(host string, relabels []*relabel.Config) *HostFilter {
 		ctx:    ctx,
 		cancel: cancel,
 
-		host: host,
+		host:     host,
+		relabels: relabels,
 
 		outputCh: make(chan map[string][]*targetgroup.Group),
 	}
 	return f
 }
 
+// PatchSD patches services discoveries to optimize performance for host
+// filtering. The discovered targets will be pruned to as close to the set
+// that HostFilter will output as possible.
+func (f *HostFilter) PatchSD(scrapes []*config.ScrapeConfig) {
+	for _, sc := range scrapes {
+		for _, d := range sc.ServiceDiscoveryConfigs {
+			switch d := d.(type) {
+			case *kubernetes.SDConfig:
+				if d.Role == kubernetes.RolePod {
+					d.Selectors = []kubernetes.SelectorConfig{{
+						Role:  kubernetes.RolePod,
+						Field: fmt.Sprintf("spec.nodeName=%s", f.host),
```

Comment:
Do we need to do any escaping here or is f.host input considered safe?

Reply:
Hmm, good question. It should be safe here, but I'll do a little investigation before merging.